lib/model/queue.ex

```diff
@@ -35,6 +35,41 @@ defmodule Queue do
       def handle_call(:get, _from, state) do
         { :reply, state, state }
       end
+
+      defp add_new_element(state, new_message) do
+        new_element = %{message: new_message, consumers_that_did_not_ack: [], number_of_attempts: 0}
+        Map.put(state, :elements, [new_element|state.elements])
+      end
+
+      defp delete_element(state, element) do
+        Map.put(state, :elements, List.delete(state.elements, element))
+      end
+
+      defp update_specific_element(state, message, update_element) do
+        Map.put(state, :elements, Enum.map(state.elements, fn element ->
+          if element.message == message do update_element.(element) else element end
+        end))
+      end
+
+      defp add_subscriber(state, subscriber) do
+        Map.put(state, :subscribers, [subscriber|state.subscribers])
+      end
+
+      defp remove_subscriber(state, subscriber) do
+        Map.put(state, :subscribers, List.delete(state.subscribers, subscriber))
+      end
+
+      defp update_consumers_that_did_not_ack(element, consumer_pid) do
+        Map.put(element, :consumers_that_did_not_ack, List.delete(element.consumers_that_did_not_ack, consumer_pid))
+      end
+
+      defp init_element(element, subscribers) do
+        element
+          |> Map.put(:consumers_that_did_not_ack, subscribers)
+          |> increase_number_of_attempts
```

Comment:
No entendi que seria lo qe hace este metodo?

Reply:
Se usa acá:
```elixir
defp add_receivers_to_state_message(state, subscribers, message) do
    send_to_replica(state.name, { :add_receivers_to_state_message, subscribers, message })
    update_specific_element(state, message, &(init_element(&1, subscribers)))
end
```
Cuando desde la cola le mandas el mensaje a todos los suscriptores, en esa funcion `init_element` guardas para ese elemento en particular, todos los suscriptores a los que le acabas de mandar el mensaje, y seteas la cantidad de veces que mandaste ese mensaje en 1 (despues cuando se reintenta se va incrementando ese `number_of_attempts`)

Reply:
Quizas no es el mejor nombre del mundo, podria ser algo como `init_sent_element_props` o algo asi, porque como que inicializas los suscriptores y el number of attempts de ese mensaje que acabas de enviar

Reply:
Sip quizas cambiaria el nombre del metodo como para que sea mas descriptivo